### ISSUE TITLE
Generate random number in cuda kernel

### ISSUE THREAD

**Issue by JuewenPeng** [sniklaus/softmax-splatting#58]
Really nice work. 

I want to ask a technical problem. Is it possible to generate random number in cuda kernel. If you could give me some suggestions, I would really appreciate that.

Looking forward to your reply.

**Comment by sniklaus** [comment]
I am under the impression that generating a random number inside of a CUDA kernel isn't as straightforward as one may think. I would recommend you generate the random numbers in PyTorch and then pass the tensor to the CUDA kernel as an additional input.

**Comment by JuewenPeng** [comment]
Thank you very much
